### PR TITLE
Pulse test button

### LINKED ISSUE #1571: Provide a "Test Now" button for sending a pulse out immediately

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #1571

### CODE REVIEW

File: frontend/src/services.js [hunk 1]
```diff
@@ -623,6 +623,12 @@ CoreServices.factory('Pulse', ['$resource', '$cookies', function($resource, $coo
             params: { pulseId: '@pulseId' },
             headers: { 'X-CSRFToken': () => $cookies.csrftoken },
         },
+        test: {
+            url: '/api/pulse/test',
+            method: 'POST',
+            params: { pulseId: '@id' },
```
Comment (agilliland) [inline frontend/src/services.js]: i think we should have this be at `/api/pulse/:id/test` which has been our normal convention for endpoints that represent a specific object.

File: frontend/src/services.js [hunk 2]
```diff
@@ -623,6 +623,12 @@ CoreServices.factory('Pulse', ['$resource', '$cookies', function($resource, $coo
...
+            headers: { 'X-CSRFToken': () => $cookies.csrftoken },
```
Comment (agilliland) [inline frontend/src/services.js]: we can delete all that old CSRFToken stuff, that was specific to the django app.  i removed all of these from the other endpoints a little while ago.

File: frontend/src/services.js (hunk 1 above)
Comment (tlrobinson) [inline frontend/src/services.js]: This endpoint is for unsaved pulses so we can test unsaved changes. I don't think it makes sense to include the id in the URL (we can also remove the `pulseId` param)

**Comment by agilliland** [comment]
@mazameli @kdoh we should check the pattern we are using here and make sure we are being consistent.

not sure what was discussed on this, but the email settings provide a "send test now" button which is specifically designed to work only after you've saved your settings.  i think it makes more sense to be consistent about that in places where there are buttons for "test now".

**Comment by tlrobinson** [comment]
I think the thinking was you'd use the test button to make sure it worked/looked correctly in a particular channel before you save it.

Also we'd have to rethink redirecting back to `/pulse` when saving, since that would be rather obnoxious.

**Comment by tlrobinson** [comment]
Ok this should be ready to go. I also made some changes to validation, so play around with that to see if it feels right.

**Comment by mazameli** [comment]
@agilliland, to recap a conversation we just had, I don't recall specifying on the SMTP settings page that it was required that the user save their settings before being able to send a test email. I think that was a miscommunication. If the concern is that an admin might test the email before saving, and then forget to save, and leave the settings page, then the better way to solve that specific problem is to present a warning modal, informing the user that they have unsaved changes.

**Pull request merged**